packages/geoadmin-elevation-profile/src/profile.api.ts

```diff
@@ -0,0 +1,349 @@
+        log.error('Malformed chunk', chunk)
+        throw new ElevationProfileError('profile_network_error', new Error('Malformed chunk'))
+    }
+    return [
+        ...chunk.coordinates.map((coordinate) => {
+            let dist = lastDist
+            if (lastCoordinate) {
+                dist += Math.sqrt(
+                    Math.pow(lastCoordinate[0] - coordinate[0], 2) +
+                        Math.pow(lastCoordinate[1] - coordinate[1], 2)
+                )
+            }
+            lastDist = dist
+            lastCoordinate = coordinate
+            const point: ServiceProfilePoints = {
+                easting: coordinate[0],
+                northing: coordinate[1],
+                dist: outputProjection.roundCoordinateValue(dist),
+                alts: undefined,
+            }
+            return point
+        }),
+    ]
+}
+
+function ensureDoubleNestedArray(
+    arr: SingleCoordinate[] | SingleCoordinate[][]
+): SingleCoordinate[][] {
+    if (Array.isArray(arr) && Array.isArray(arr[0]) && Array.isArray(arr[0][0])) {
+        if (typeof arr[0][0] !== 'number') {
+            throw new ElevationProfileError(
+                'profile_could_not_generate',
+                new Error(
+                    'Received a multi-feature (MultiLineString or MultiPolygon). This is not supported by this component, you need to split the feature and give each element of the "multi"-feature separately.'
+                )
+            )
+        }
+        return arr as SingleCoordinate[][]
+    }
+    return [arr] as SingleCoordinate[][]
+}
+
+/**
+ * Gets profile from https://api3.geo.admin.ch/services/sdiservices.html#profile
+ *
+ * @param profileCoordinates Coordinates, expressed in the given projection, from which we want the
+ *   profile
+ * @param projection The projection used to describe the coordinates
+ * @returns The profile, or null if there was no valid data to produce a profile
+ * @throws ElevationProfileError will reject the promise with a ProfileError if something went
+ *   wrong.
+ */
+export default async (
+    profileCoordinates: SingleCoordinate[] | SingleCoordinate[][],
+    projection: CoordinateSystem,
+    staging: Staging = 'production'
+): Promise<ElevationProfile> => {
+    if (!profileCoordinates || profileCoordinates.length === 0) {
+        const errorMessage = `Coordinates not provided`
+        log.error(errorMessage)
+        throw new ElevationProfileError(errorMessage, new Error('profile_could_not_generate'))
+    }
+    const segments: ElevationProfileSegment[] = []
+    // if the profileCoordinates is not a double nested array, we make it one
+    // segmented files have a double nested array, but not all files or self made drawings
+    // so we have to make sure we have a double nested array and then iterate over it
+    const coordinatesForSegments: SingleCoordinate[][] = ensureDoubleNestedArray(profileCoordinates)
+
+    for (const coordinates of coordinatesForSegments) {
```

Comment:
Maybe you could refactor this chunk into several smaller functions to increase understandability